src/kibana/plugins/discover/components/field_chooser/field_chooser.js

```diff
@@ -128,7 +128,7 @@ define(function (require) {
           var type = isGeoPoint ? 'tile_map' : 'histogram';
           // If we're visualizing a date field, and our index is time based (and thus has a time filter),
           // then run a date histogram
-          if (field.type === 'date' && $scope.indexPattern.timeFieldName) {
+          if (field.type === 'date' && field.indexed && $scope.indexPattern.timeFieldName) {
```

Comment:
what you want is `field.type === 'date' && $scope.indexPattern.timeFieldName === field.name`